packages/react-components/src/components/ResponsiveVerticalGallery.tsx

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IStyle, mergeStyles } from '@fluentui/react';
+import { _convertRemToPx } from '@internal/acs-ui-common';
+import React, { useRef } from 'react';
+import { _useContainerHeight } from './utils/responsive';
+import { VerticalGallery, VerticalGalleryStyles } from './VerticalGallery';
+
+/**
+ * Props for the Responsive wrapper of the VerticalGallery component
+ *
+ * @beta
+ */
+export interface ResponsiveVerticalGalleryProps {
+  /** Video tiles to be rendered in the Vertical Gallery */
+  children: React.ReactNode;
+  /** Styles for the Children space container */
+  containerStyles: IStyle;
+  /** Styles for the VerticalGallery component */
+  verticalGalleryStyles: VerticalGalleryStyles;
+  /** Height of the gap in between the video tiles */
+  gapHeightRem: number;
+  /** Height of the control bar for navigating pages */
+  controlBarHeightRem?: number;
+}
+
+/**
+ * Responsive container for the VerticalGallery Component. Performs calculations for number of children
+ * for the VerticalGallery
+ * @param props
+ *
+ * @beta
+ */
+export const ResponsiveVerticalGallery = (props: ResponsiveVerticalGalleryProps): JSX.Element => {
+  const { children, containerStyles, verticalGalleryStyles, gapHeightRem, controlBarHeightRem } = props;
+  const containerRef = useRef<HTMLDivElement>(null);
+  const containerHeight = _useContainerHeight(containerRef);
+
+  const topPadding = containerRef.current ? parseFloat(getComputedStyle(containerRef.current).paddingTop) : 0;
+  const bottomPadding = containerRef.current ? parseFloat(getComputedStyle(containerRef.current).paddingBottom) : 0;
+
+  const childrenPerPage = calculateChildrenPerPage({
+    numberOfChildren: React.Children.count(children),
+    containerHeight: (containerHeight ?? 0) - topPadding - bottomPadding,
+    gapHeightRem,
+    controlBarHeight: controlBarHeightRem ?? 2
+  });
+  return (
+    <div ref={containerRef} className={mergeStyles(containerStyles)}>
+      <VerticalGallery childrenPerPage={childrenPerPage} styles={verticalGalleryStyles}>
+        {children}
+      </VerticalGallery>
+    </div>
+  );
+};
+
+/**
+ * Helper function to find the number of children for the VerticalGallery on each page.
+ */
+const calculateChildrenPerPage = (args: {
```

Comment:
this is the kinda function we should really have a unit test for... (or a unit test for - given these children and this size of vertical gallery I expect `n` children and `m` pages) but I wouldn't block PR on it